### PR TITLE
[7.17] (backport elastic-agent#680) Fix transpiler to allow : in dynamic variables.

### PR DESCRIPTION
**Pull request by michel-laterman**
Backport https://github.com/elastic/elastic-agent/pull/680

Fix transpiler regex to allow ':' characters in dynamic variables so
that users can input "${dynamic.lookup|'fallback.here'}".

### CODE REVIEW

**Comment by AndersonQ** [comment]
@michel-laterman I believe it'd be better for the same reviewers of the original PR (https://github.com/elastic/elastic-agent/pull/680) to review this backport.

File: x-pack/elastic-agent/pkg/agent/transpiler/vars.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/elastic/beats/v7/libbeat/common"
 )
 
-var varsRegex = regexp.MustCompile(`\${([\p{L}\d\s\\\-_|.'"]*)}`)
+var varsRegex = regexp.MustCompile(`\${([\p{L}\d\s\\\-_|.'":]*)}`)
```
Comment (ph) [inline x-pack/elastic-agent/pkg/agent/transpiler/vars.go:18]: Small comment here, this mean the following variable declaration would be valid:
```
${dynami:c.lookup|'fallback.here'}
```
`dynami:c.lookup` seems weird to me, but again... this is still a valid key reference in YAML if you quote the key value.
```yaml
'test:ok': hello
```

**Review by ph — APPROVED** [review]

File: x-pack/elastic-agent/pkg/agent/transpiler/vars.go (hunk 1 above)
Comment (michel-laterman) [reply x-pack/elastic-agent/pkg/agent/transpiler/vars.go:18]: yeah! yaml? 😕

**Comment by michel-laterman** [comment]
e2e tests in https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackaging/detail/PR-32407/4/pipeline/ passed

File: x-pack/elastic-agent/pkg/agent/transpiler/vars.go (hunk 1 above)
Comment (ph) [reply x-pack/elastic-agent/pkg/agent/transpiler/vars.go:18]: I had yaml with a passion, not sure it was better than xml ;)

**Pull request merged**